src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2342,7 +2343,7 @@ private BoundExpression BindUnaryOperatorCore(CSharpSyntaxNode node, string oper
                     constantValueOpt: ConstantValue.NotAvailable,
                     methodOpt: null,
                     resultKind: LookupResultKind.Viable,
-                    type: operand.Type);
+                    type: operand.Type!);
```

Comment:
consider using `RoslynDebug.Assert(operand.Type is object)`

Reply:
We're inside an `if` that tested that this operand has type `dynamic`. I don't think it's important to add another assert for this.